cypress/support/commands.js

```diff
@@ -250,8 +248,102 @@ Cypress.Commands.add('checkElementComponentContainsValue', (mainSelector, compon
+ * @param {String} filename File path (with its root at the directory containing the cypress.json file)
+ * @param {String} hostname Host name for OpenSearch
+ * @param {String} port Port for OpenSearch
+ */
+
+Cypress.Commands.add('importJSONMapping', (filepath, hostname = 'localhost', port = '9200') => {
+  cy.readFile(filepath, 'utf8').then((str) => {
+    const strSplit = str.split('\n\n')
+    cy.wrap(strSplit).each((element) => {
+      const json = JSON.parse(element)
+      if (json.type === 'index') {
+        const index = json.value.index
+        const settings = json.value.settings
+        const mappings = json.value.mappings
+        const aliases = json.value.aliases
+        const body = { settings, mappings, aliases }
+        cy.request({ method: 'PUT', url: hostname + ':' + port + '/' + index, body: body, failOnStatusCode: false, log: true }).then((response) => {
+        
+        })
+      }
     })
+  })
+})
+
+/**
+ * Read indices from a file and request for them to be deleted from OpenSearch using the delete index API
+ * @param {String} filename File path (with its root at the directory containing the cypress.json file)
+ * @param {String} hostname Host name for OpenSearch
+ * @param {String} port Port for OpenSearch
+ */
+
+Cypress.Commands.add('clearJSONMapping', (filename, hostname = 'localhost', port = '9200') => {
+  cy.readFile(filename, 'utf8').then((str) => {
+    const strSplit = str.split('\n\n')
+    cy.wrap(strSplit).each((element) => {
+      const json = JSON.parse(element)
+      if (json.type === 'index') {
+        const index = json.value.index
+        cy.request({ method: 'DELETE', url: hostname + ':' + port + '/' + index, failOnStatusCode: false, log: true }).then((response) => {
+        })
+      }
+    })
+  })
+})
+
+/**
+ * Read docs from a file and import them to OpenSearch using the bulk API
+ * @param {String} filename File path (with its root at the directory containing the cypress.json file)
+ * @param {String} hostname Host name for OpenSearch
+ * @param {String} port Port for OpenSearch
+ */
+
+Cypress.Commands.add('importJSONDoc', (filename, hostname = 'localhost', port = '9200', bulkMax = 1600) => {
+  cy.readFile(filename, 'utf8').then((str) => {
+    let line = 0
+    let bucket = 0
+    let bulkLines = [[]]
+    str.split('\n\n').forEach((element) => {
+      const json = JSON.parse(element)
+
+      const id = json.value.id
+      const index = json.value.index
+      const source = json.value.source
+
+      const body = { "index": {"_id": id, "_index": index }  }
+      const oneLineBody = JSON.stringify(body).replace('\n', '')
+      const oneLineSource = JSON.stringify(source).replace('\n', '')
+      const oneLineDoc = oneLineBody + '\n' + oneLineSource
+      bulkLines[0].push(oneLineDoc)
+      
+      line = line + 1
+      if(line % bulkMax === 0) {
+        console.log(bucket)
+        console.log("Before response" + bucket)
```

Comment:
Remove debugging statements for development.

Reply:
The statements have been cut.